Revert DaemonSet update.

madhusudancs opened the pull request:
Comment out the type fields and remove the code that uses these fields.

This is part of the DaemonSet beta graduation plan as described in https://github.com/kubernetes/kubernetes/issues/15310#issuecomment-177062027.

madhusudancs committed:
Comment out DaemonSet update type fields and remove the code that depends on it.

Leaving the type fields as comments for reference and reminder. But
deleting the conversion, defaulting and validation code. They can
always be brough back from the previous PR once the types are
introduced. Because builds break without them anyway that serves as a
reminder, so there is no need to leave them commented out.

bgrant0607 commented:
cc @mikedanese @davidopp @janetkuo @kargakis

bgrant0607 commented on pkg/apis/extensions/types.go:
```diff
@@ -353,6 +353,7 @@ type DeploymentList struct {
 	Items []Deployment `json:"items"`
 }
 
+/* Commenting out for v1.2. We are planning to bring these types back with a more robust DaemonSet update implementation in v1.3, hence not deleting but just commenting the types out.
```
Please add TODO comments

bgrant0607 replied:
To all the commented-out sections.

madhusudancs replied:
Done.

bgrant0607 commented:
ValidateDaemonSetTemplateUpdate still prohibits updates. @madhusudancs Do you plan to change that in another PR?

madhusudancs commented:
> ValidateDaemonSetTemplateUpdate still prohibits updates. @madhusudancs Do you plan to change that in another PR?

Yeah, my idea was to (and still is to) address each bullet point you proposed here - https://github.com/kubernetes/kubernetes/issues/15310#issuecomment-177062027 in a separate PR.

bgrant0607 commented:
LGTM

madhusudancs commented:
@bgrant0607 Addressed the review comments. PTAL.

I will send separate PRs for ValidateDaemonSetTemplateUpdate() and enabling DaemonSet by default.

(sorry, this was sitting in draft).

madhusudancs commented:
Just squashed the commits. Adding LGTM back.

madhusudancs commented:
@k8s-bot test this please.

madhusudancs commented:
@k8s-bot test this issue: #21045

madhusudancs commented:
@k8s-bot test this issue: #21045

bgrant0607 commented:
@mikedanese Could we please get a manual merge on this?

eparis commented:
note that adding `priority/p*` would bump it up the submit queue.  Usually p2 jumps straight to the head of the line.

bgrant0607 commented:
Thanks @eparis. Didn't know that.

Thanks @mikedanese
